code/core/system/proxy-velodyne32/include/velodyne32Decoder.h

```diff
@@ -0,0 +1,139 @@
+     */  
+    Velodyne32Decoder(const std::shared_ptr< SharedMemory > m,
+odcore::io::conference::ContainerConference &c, const string &s, const bool &withCPC, const uint8_t &SPCOption, const uint8_t &CPCIntensityOption, const uint8_t &numberOfBitsForIntensity, const uint8_t &intensityPlacement, const uint8_t &distanceEncoding);
+    
+    /**
+         * Constructor.
+     * @param c container conference 
+     * @param s name of the calibration file
+     * @param CPCIntensityOption with or without intensity
+     * @param numberOfBitsForIntensity number of bits reserved for intensity
+     * @param intensityPlacement higher or lower bits for intensity
+     * @param distanceEncoding use cm or 2mm for distance encoding
+     */
+    Velodyne32Decoder(odcore::io::conference::ContainerConference &c, const string &s, const uint8_t &CPCIntensityOption, const uint8_t &numberOfBitsForIntensity, const uint8_t &intensityPlacement, const uint8_t &distanceEncoding);
+
+    virtual ~Velodyne32Decoder();
+
+    virtual void nextString(const std::string &s);
+
+   private:
+    void readCalibrationFile();
+    void index32sensorIDs();
+    void setupIntensityMaskCPC(uint8_t &, uint8_t &);
+    void sendCPC(bool);
+    void sendPointCloud();
+   private:
+    const uint32_t m_MAX_POINT_SIZE = 70000; //the maximum number of points per frame. This upper bound should be set as low as possible, as it affects the shared memory size and thus the frame updating speed.
+    const uint32_t m_SIZE_PER_COMPONENT = sizeof(float);
+    const uint8_t m_NUMBER_OF_COMPONENTS_PER_POINT = 4;  //4 components per vector: (1) cartesian: xyz+intensity; (2) polar: distance+azimuth+vertical angle+intensity
+    const uint32_t m_SIZE = m_MAX_POINT_SIZE * m_NUMBER_OF_COMPONENTS_PER_POINT * m_SIZE_PER_COMPONENT; //the total size of the shared memory
+
+    uint8_t m_SPCOption; //0: xyz+intensity; 1: distance+azimuth+vertical angle+intensity
+    uint8_t m_CPCIntensityOption; //Only used when CPC is enabled. 0: without intensity; 1: with intensity; 2: send a CPC container twice, one with intensity, and the other without intensity
+    uint8_t m_numberOfBitsForIntensity; //Range 0-7. Only used when CPC is enabled
+    uint8_t m_intensityPlacement;  //0: higher bits; 1: lower bits
+    uint16_t m_mask;  //for combining distance and intensity in 16 bits
+    uint8_t m_distanceEncoding; //0: cm; 1: 2mm
+    uint32_t m_pointIndexSPC; //current number of points of the current frame for shared point cloud 
+    uint32_t m_pointIndexCPC; //current number of points of the current frame for compact point cloud
+    uint32_t m_startID;
+    float m_previousAzimuth;
+    float m_currentAzimuth;
+    float m_distance;
+    std::shared_ptr< SharedMemory > m_velodyneSharedMemory; //shared memory for shared point cloud
+    float *m_segment;  //temporary memory for transferring data of each frame to the shared memory
+    odcore::io::conference::ContainerConference &m_velodyneContainer;
```

Comment:
Can you please refactor this name from m_velodyneContainer to m_conference as it would help a script in OpenDLV to automatically extract the data flow between the software modules.